Fix #4658

mnakalay opened the pull request:
@mlocati rightfully pointed out that when using a block's task we needed a way to know which instance of the block launched the call.

He submitted a solution that worked in some situations but not all.

For instance, it broke the topic list block.

This pull request builds on top of his solution.

@mlocati solution only worked when the check on the task was done by the block's controller. In that case the checking function automatically adds the bID to the parameters and @mlocati check works.

But sometimes, the task will first go through the page's controller.

I modified the task checking function in the page's controller to also add the bID to the list of parameters before running the check. If the check is succesful, the bID is removed from parameters and the task is processed.

mnakalay commented:
Hi @a3020 
1- Because this is a fix for #4658 and there's a long discussion/debate over there, I thought I didn't need to explain here but you're right I should probably add something to that effect
2- the [NOUR] is absolutely not necessary, it's just an old habit of mine that makes it easy for me to find my own modifications. I just forgot that git makes it easy as well. I am not sure how to deal with it now, though. Can I amend my pull request or should I submit a new one?

a3020 commented:
Yeah, you can amend and then force push to your branch. It will automatically update this PR.

mnakalay commented:
ok I modified my description and removed my name. I think I did it correctly :)

In the bug report debate, @aembler ended up with a list of tests that needed to be done to make sure this works. I'm going to start on that but if others could do it too, just for confirmation, that would be great.

They are:

1. Built-in Blog List page, filtering page titles, page list and the topics list properly.
2. Form Submit
3. Survey Submit.
4. Form submission when block is in a stack.
5. Form submission when a block is in a clipboard
6. Form submission when multiple forms are on a page, only the proper form submits and shows the message.

mnakalay commented:
for some reason that I don't understand all travis check fail on my last commits so I'm closing this and sending a new pull request
